src/DataObject/Controller/CloneController.php

```diff
@@ -0,0 +1,105 @@
+use Pimcore\Bundle\StudioBackendBundle\Exception\Api\ForbiddenException;
+use Pimcore\Bundle\StudioBackendBundle\Exception\Api\NotFoundException;
+use Pimcore\Bundle\StudioBackendBundle\Exception\Api\UserNotFoundException;
+use Pimcore\Bundle\StudioBackendBundle\OpenApi\Attributes\Parameters\Path\IdParameter;
+use Pimcore\Bundle\StudioBackendBundle\OpenApi\Attributes\Response\Content\IdJson;
+use Pimcore\Bundle\StudioBackendBundle\OpenApi\Attributes\Response\CreatedResponse;
+use Pimcore\Bundle\StudioBackendBundle\OpenApi\Attributes\Response\DefaultResponses;
+use Pimcore\Bundle\StudioBackendBundle\OpenApi\Attributes\Response\SuccessResponse;
+use Pimcore\Bundle\StudioBackendBundle\OpenApi\Config\Tags;
+use Pimcore\Bundle\StudioBackendBundle\Util\Constants\ElementTypes;
+use Pimcore\Bundle\StudioBackendBundle\Util\Constants\HttpResponseCodes;
+use Pimcore\Bundle\StudioBackendBundle\Util\Constants\UserPermissions;
+use Pimcore\Bundle\StudioBackendBundle\Util\Traits\PaginatedResponseTrait;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Attribute\MapRequestPayload;
+use Symfony\Component\Routing\Attribute\Route;
+use Symfony\Component\Security\Http\Attribute\IsGranted;
+use Symfony\Component\Serializer\SerializerInterface;
+
+/**
+ * @internal
+ */
+final class CloneController extends AbstractApiController
+{
+    use PaginatedResponseTrait;
+
+    public function __construct(
+        SerializerInterface $serializer,
+        private readonly CloneServiceInterface $cloneService,
+    ) {
+        parent::__construct($serializer);
+    }
+
+    /**
+     * @throws AccessDeniedException
+     * @throws ElementSavingFailedException
+     * @throws ForbiddenException
+     * @throws NotFoundException
+     * @throws UserNotFoundException
+     */
+    #[Route('/data-objects/{id}/clone/{parentId}', name: 'pimcore_studio_api_data_objects_clone', methods: ['POST'])]
+    #[IsGranted(UserPermissions::DATA_OBJECTS->value)]
+    #[Post(
+        path: self::API_PATH . '/data-objects/{id}/clone/{parentId}',
+        operationId: 'cloneDataObject',
+        summary: 'Clone a specific data object.',
+        tags: [Tags::DataObjects->value]
+    )]
+    #[SuccessResponse(
+        description: 'Successfully copied data object',
+    )]
+    #[CreatedResponse(
+        description: 'Successfully copied parent data object and created jobRun for copying child elements',
+        content: new IdJson('ID of created jobRun')
+    )]
+    #[IdParameter(type: ElementTypes::TYPE_DATA_OBJECT)]
+    #[IdParameter(type: ElementTypes::TYPE_DATA_OBJECT, name: 'parentId')]
+    #[CloneRequestBody]
+    #[DefaultResponses([
+        HttpResponseCodes::UNAUTHORIZED,
+        HttpResponseCodes::NOT_FOUND,
+    ])]
+    public function cloneDataObject(
+        int $id,
+        int $parentId,
+        #[MapRequestPayload] CloneParameters $parameters
+    ): Response {
+        $status = HttpResponseCodes::SUCCESS->value;
+        $data = null;
```

Comment:
Does data have a purpose here?
It is always going to be null no?

Reply:
Good point, we can clean this up :)